lang/LangPrimSource/PyrListPrim.cpp

```diff
@@ -44,6 +44,8 @@ int class_identdict_index, class_identdict_maxsubclassindex;
 PyrClass *class_identdict;
 PyrSymbol *s_proto, *s_parent;
 PyrSymbol *s_delta, *s_dur, *s_stretch;
+PyrSymbol *s_type, *s_rest, *s_empty, *s_r;
+PyrClass *class_rest, *class_metarest;
```

Comment:
You explained that they should not be added to PyrObject as intrinsic classes. Here, they are added to PyrListPrim and I am not adding any intrinsic classes.

Note that I am imitating a speed optimization that has been present in prEvent_Delta for some 15 years now. If you feel it's inappropriate, you might contact James McCartney about that.

Also, the speed improvement in my benchmarks was quite dramatic. It is really wasteful to look up these symbols on every invocation of isRest.